cmake/CMakeLists.txt

```diff
@@ -1140,6 +1148,13 @@ if (onnxruntime_USE_COREML)
     list(APPEND ORT_PROVIDER_CMAKE_FLAGS -Donnxruntime_USE_COREML=1)
     list(APPEND ONNXRUNTIME_PROVIDER_NAMES coreml)
 endif()
+
+add_subdirectory(external/tracy EXCLUDE_FROM_ALL)
+include_directories(external/tracy)
+if(onnxruntime_ENABLE_TRACY)
+    add_compile_definitions(TRACY_ENABLE)
+endif()
```

Comment:
Not exactly. Tracy defines empty macro so that manual insturment profiling can be enabled/disabled easily. FYI, see [Tracy.hpp](https://github.com/wolfpld/tracy/blob/0b1e5d2c9666ce7810447b6bc53026ab3a5cb650/Tracy.hpp#L7-L18 )
so add_subdirectory and include_directories should be configured regardless of enabling of tracy.

This is added in top level is because I added some FrameMark in perftest tool. If it is not desired for other EP, I can move it to OpenCL EP only, tho.

Reply:
Not quite following. My question was more about the usage of Tracy in general and not whether it was enabled or disabled in that usage. i.e. Wouldn't any code attempting to use Tracy be inside USE_OPENCL ifdefs so we don't need to add it to the build unless OpenCL is enabled?

Reply:
OK, It is in fact I am using the very mechanism provided by Tracy itself, without bothering wrap all Tracy macro usages with `ifdef`s. If `TRACY_ENABLE` is not define, all those macro defined in `Tracy.hpp` is empty automatically. And it happens that I used some of those macros in `onnxruntime/test/perftest` files. 

So to leverage the convenience provided by tracy developer, `Tracy.hpp` is added as a top level include. But the linking only happens if `USE_OPENCL` and `TRACY_ENABLE`

Reply:
So for tracy, enable is equivalent to use.

Reply:
I guess it's okay for now but this should probably be cleaned up before checkin to master. You're forcing all builds to install Tracy and go via those headers (even though there's no linking) despite Tracy only being relevant to OpenCL. It also makes it slightly more obtuse to understand what code in performance_runner is doing as you need to find where things like ZoneScopedN is to see the macro is a no-op.